base/libgit2/callbacks.jl

```diff
@@ -28,16 +28,8 @@ function authenticate_ssh(creds::SSHCredentials, libgit2credptr::Ptr{Ptr{Void}},
         username_ptr, schema, host)
     isusedcreds = checkused!(creds)
 
-    errcls, errmsg = Error.last_error()
-    if errcls != Error.None
-        # Check if we used ssh-agent
-        if creds.usesshagent == "U"
-            creds.usesshagent = "E" # reported ssh-agent error, disables ssh agent use for the future
```

Comment:
In the [logic that controls the ssh-agent support](https://github.com/JuliaLang/julia/pull/20738/files/b5dfcc7fb095f9ae843777af02771c7723dfb775#diff-43fbc9e635c9a9f0da9707c61e66dea7R32) you could run into an infinite loop if `git_cred_ssh_key_from_agent` would always return 0. To avoid this I dropped `creds.usesshagent == "U"` which made this logic unnecessary.